server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalVariableWidthHistogram.java

```diff
@@ -451,7 +450,8 @@ private void mergeBucketsWithPlan(List<Bucket> buckets, List<BucketRange> plan,
             }
             toMerge.add(buckets.get(startIdx)); // Don't remove the startIdx bucket because it will be replaced by the merged bucket
 
-            reduceContext.consumeBucketsAndMaybeBreak(- (toMerge.size() - 1));
+            int toRemove = toMerge.stream().mapToInt(b -> countInnerBucket(b)+1).sum();
```

Comment:
Is this an actual bug you found while making this change?

Reply:
A minor one, yes. The max bucket count is not accurate when buckets are auto-merged.